packages/rest-api-client/src/KintoneRequestConfigBuilder.ts

```
@@ -46,37 +48,44 @@ export class KintoneRequestConfigBuilder implements RequestConfigBuilder {
         pfxFilePath: string;
         password: string;
       };
-  private proxy?: ProxyConfig;
+  private readonly proxy?: ProxyConfig;
   private requestToken: string | null;
 
-  constructor({
-    baseUrl,
-    auth,
-    basicAuth,
-    clientCertAuth,
-    proxy,
-    userAgent,
-  }: {
-    baseUrl: string;
-    auth: DiscriminatedAuth;
-    basicAuth?: BasicAuth;
-    clientCertAuth?:
+  constructor(
+    options: {
+      baseUrl: string;
+      auth: DiscriminatedAuth;
+      basicAuth?: BasicAuth;
+      proxy?: ProxyConfig;
+      userAgent?: string;
+    } & (
+      | { httpsAgent: https.Agent }
       | {
-          pfx: Buffer;
-          password: string;
+          clientCertAuth?:
+            | {
+                pfx: Buffer;
+                password: string;
+              }
+            | {
+                pfxFilePath: string;
+                password: string;
+              };
         }
-      | {
-          pfxFilePath: string;
-          password: string;
-        };
-    proxy?: ProxyConfig;
-    userAgent?: string;
-  }) {
-    this.baseUrl = baseUrl;
-    this.auth = auth;
-    this.headers = this.buildHeaders({ basicAuth, userAgent });
-    this.clientCertAuth = clientCertAuth;
-    this.proxy = proxy;
+    )
+  ) {
+    this.baseUrl = options.baseUrl;
+    this.auth = options.auth;
+    this.headers = this.buildHeaders({
+      basicAuth: options.basicAuth,
+      userAgent: options.userAgent,
+    });
+    if ("httpsAgent" in options) {
+      this.httpsAgent = options.httpsAgent;
+    }
+    if ("clientCertAuth" in options) {
+      this.clientCertAuth = options.clientCertAuth;
+    }
```

I believe the method should throw an error when the user specifies both the parameter `clientCertAuth` and `httpAgent`.  The README says:

> When `httpsAgent` parameter is given, this parameter is *unavailable*.

But actually, the `httpsAgent` is *ignored* rather than *unavailable*.